### ISSUE TITLE
imagep(topo) should use better aspect ratio

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1008]
I would have thought this would have been done before, but the code below shows that the aspect ratio is not being set for the latitude in question. (See the `imagep` line, and try putting that aspect ratio into the call ... if you know the geometry you'll see the problem.)
```R
library(oce)
library(marmap)
## from CL email 20160630
lon0 <- -45.59
lat0 <- 48.78
b <- getNOAA.bathy(lon0-5,lon0+5,lat0-8,lat0+8,resolution=8)
lon <- as.numeric(rownames(b))
lat <- as.numeric(colnames(b))
z <- as.matrix(b)
topo <- as.topo(lon,lat,z) # oce object
imagep(topo) # asp=1/cos(lat0*pi/180))
abline(h=lat0)
abline(v=lon0)
```

**Comment by richardsc** [comment]
Good idea!

Just for fun I tried doing:
```r
mapPlot(topo)
mapImage(topo)
```
and the result was surprisingly sensible.
![screen shot 2016-06-30 at 7 01 40 pm](https://cloud.githubusercontent.com/assets/233584/16507075/10ededd4-3ef5-11e6-9643-3a41a9439544.png)

**Comment by dankelley** [comment]
Yeah, even with a funny `lon0`, the projection looks quite good. Something is funky with that graph of yours, though -- no lat labels. I'm kinda done for the day but maybe you could make a bug report on that, since it looks quite bad to me.

**Comment by richardsc** [comment]
Actually, the lack of labels I think occurred because it's a bit slow, and I interrupted the process.
```
> system.time(mapPlot(topo))
   user  system elapsed 
 24.937   1.001  25.981 
```
If you add `debug=10` you can see that it's going through all the sides looking for axis intersections ... not sure why that takes 25 seconds (on this particular 8 year old machine).

**Comment by dankelley** [comment]
The problem with slowness can probably be fixed up a bit. I don't think calculating the label positions should be too slow, but the lines will be slow because the code doesn't "know" much about the plot scale, and so it draws line segments that won't appear, and also there are a lot of segments to try to get smoothness.  Both of these are a bit tricky when writing code that should work for any projection. For a given projection its a lot easier, of course, but in general there's no way of knowing (without calculating) exactly what will be in the view port. For example, in mercator, if we find that the 30N line was wholly above the plot region, then we could skip higher latitudes. But that would fail on other projections. My choice was not to special-case things because trying to handle N cases instead of 1 case increases the coding and debugging effort by N.

Even without special casing, my guess is that these things could be speeded up quite a lot (2X or so) by some simple tweaking. I'll look at that some time but don't really want an issue for it, because I use these issues as a to-do list and this is less important than, say, the UHL problem.

**Comment by dankelley** [comment]
Fixed in `develop` commit c50d2db31ea8020ba2f6ccfad88c999b58530c4a
